vars/buildImage.groovy

```diff
@@ -0,0 +1,81 @@
+// Create a Job and BuildConfig Template to trigger the Build
+// Available parameters:
+//    dockerFile    string -- DockerFile used for the buildconfig
+//    gitURL:       string -- Required - Git URL used for the buildconfig
+//    image         string -- Image used for the buildconfig
+//    nameSpace:    string -- Namespace name to run the build
+def call(params = [:]) {
+    def imageName
+    def bcObj
+    openshift.withCluster() {
+        openshift.withProject() {
+            stage('CleanUp') {
+                // Once we have the TTL controller enabled we can remove this stage
+                // More info: https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/
+                def jobs = openshift.selector("jobs").names()
+                for (job in jobs) {
+                    if (job.startsWith('job/cleanup')) {
+                        def obj = openshift.selector(job).object()
+                        if (obj.status.completionTime) {
+                            openshift.selector(job).delete()
+                        }
+                    }
+                }
+            }
+
+            stage('Prepare Env') {
+               if (!params['gitURL']) {
+                   currentBuild.result = 'ABORTED'
+                   error("GitURL is a required field. Aborting")
+               }
+
+               def bcJSON  = libraryResource 'com/github/coreos/buildconfig.json'
+               def jobJSON = libraryResource 'com/github/coreos/job.json'
+               def jobObj  = readJSON text: jobJSON
+               bcObj       = readJSON text: bcJSON
+               def tagName = UUID.randomUUID()
+
+               jobObj      = openshift.process(jobObj, "-p", "NAME=cleanup-${tagName}")
+               def app     = openshift.create(jobObj)
+               def job     = app.narrow('job')
+               def jobMap  = job.object()
+               def uid     = jobMap.metadata.get('uid')
+
+               imageName = "bc-${tagName}"
+               //Create and unique name for image and bc
+               bcObj['parameters'][0] +=['name': 'NAME', 'value': "bc-${tagName}".toString()]
```

Comment:
Using indexes like that will be hard to maintain. That'll be good motivation too to tackle https://github.com/coreos/coreos-ci-lib/pull/61#discussion_r569401940 as a follow-up!